internal/hcl/project_locator.go

```diff
@@ -16,19 +16,20 @@ import (
 )
 
 var (
-	defaultEnvVarNames = regexp.MustCompile(`^(prd|prod|production|preprod|staging|stage|stg|development|dev|release|testing|test|tst|qa|uat|live|sandbox|demo|integration|int|experimental|experiments|trial|validation|perf|sec|dr)`)
+	defaultEnvVarNames = regexp.MustCompile(`^(.*-)?(prd|prod|production|preprod|staging|stage|stg|development|dev|release|testing|test|tst|qa|uat|live|sandbox|demo|integration|int|experimental|experiments|trial|validation|perf|sec|dr)`)
```

Comment:
Should we also add `(.*-)$` at the end of this to make sure that the env is a separate token.

As well as `-` we might also want to support `_` as a separator.